pacman/game.py

```diff
@@ -33,15 +33,43 @@ def add_ghost(self, agent):
             raise OutsideOfLegalPath("Cannot add agent\
                 to invalid board position")
 
-    def play_game(self, game_steps):
+    def check_rules(self, pacman_old_node, pacman_new_node,
+                    ghost_old_node, ghost_new_node):
+        collision = False
+        if (pacman_new_node.position == ghost_new_node.position):
+            collision = True
+        if ((pacman_new_node.position == ghost_old_node.position) and (
+                pacman_old_node.position == ghost_new_node.position)):
+            collision = True
+        return collision
```

Comment:
It is actually not that simple.
There are at least two cases where there must be an end of the game : 

- either the ghost and pacman's move are towards the same node

- or they are side by side and they move one towards the other, producing a collision (if the game did not terminate, they would exchange places)

Hence the extraction of the rules in its own function.

An alternative would be to make the ghost move before pacman but this would give an extra information to the pacman about where the ghost's are, which might give it an edge in learning.